backend/src/layer/logic/image_review/test/image_review_database_mock.rs

```diff
@@ -0,0 +1,139 @@
+pub struct ImageReviewDatabaseMock {
+    n_images_by_rank_date_calls: Arc<Mutex<u32>>,
+    n_images_next_week_by_rank_not_checked_last_week_calls: Arc<Mutex<u32>>,
+    n_images_by_last_checked_not_checked_last_week_calls: Arc<Mutex<u32>>,
+    delete_image_calls: Arc<Mutex<u32>>,
+    mark_as_checked_calls: Arc<Mutex<u32>>,
+}
+
+impl ImageReviewDatabaseMock {
+    #[must_use]
+    pub fn get_n_images_by_rank_date_calls(&self) -> u32 {
+        *self
+            .n_images_by_rank_date_calls
+            .lock()
+            .expect("failed to lock mutex for `n_images_by_rank_date_calls` counter")
+    }
+    #[must_use]
+    pub fn get_n_images_next_week_by_rank_not_checked_last_week_calls(&self) -> u32 {
+        *self.n_images_next_week_by_rank_not_checked_last_week_calls.lock().expect("failed to lock mutex for `n_images_next_week_by_rank_not_checked_last_week_calls` counter")
+    }
+    #[must_use]
+    pub fn get_n_images_by_last_checked_not_checked_last_week_calls(&self) -> u32 {
+        *self.n_images_by_last_checked_not_checked_last_week_calls.lock().expect("failed to lock mutex for `n_images_by_last_checked_not_checked_last_week_calls` counter")
+    }
+    #[must_use]
+    pub fn get_delete_image_calls(&self) -> u32 {
+        *self
+            .delete_image_calls
+            .lock()
+            .expect("failed to lock mutex for `delete_image_calls` counter")
+    }
+    #[must_use]
+    pub fn get_mark_as_checked_calls(&self) -> u32 {
+        *self
+            .mark_as_checked_calls
+            .lock()
+            .expect("failed to lock mutex for `mark_as_checked_calls` counter")
+    }
+}
+
+#[async_trait]
+impl ImageReviewDataAccess for ImageReviewDatabaseMock {
+    /// Returns the first n images sorted by rank which are related to an meal served at the given day.
+    async fn get_n_images_by_rank_date(&self, n: u32, _date: Date) -> Result<Vec<Image>> {
+        *self
+            .n_images_by_rank_date_calls
+            .lock()
+            .expect("failed to lock mutex for `n_images_by_rank_date_calls` counter") += 1;
+
+        let mut images = Vec::new();
+        for _i in 0..n {
+            images.push(Image {
+                id: Uuid::default(),
+                image_hoster_id: String::default(),
+                url: String::default(),
+                rank: 0.0,
+                upvotes: 0,
+                downvotes: 0,
+            });
+        }
+        Ok(images)
+    }
+    /// Returns the first n images sorted by rank which are related to an meal served in the next week or which were not validated last week.
+    async fn get_n_images_next_week_by_rank_not_checked_last_week(
+        &self,
+        n: u32,
+    ) -> Result<Vec<Image>> {
+        *self.n_images_next_week_by_rank_not_checked_last_week_calls.lock().expect("failed to lock mutex for `n_images_next_week_by_rank_not_checked_last_week_calls` counter") += 1;
+
+        let mut images = Vec::new();
+        for _i in 0..n {
```

Comment:
general advice: if you dont use a variable, you can just name it `_` to make clear you discard this information